Source/Hurl.RulesetManager/MainWindow.xaml

```diff
@@ -0,0 +1,119 @@
+    xmlns:ui="http://schemas.lepo.co/wpfui/2022/xaml"
+    Title="MainWindow"
+    Width="800"
+    Height="450"
+    ExtendsContentIntoTitleBar="True"
+    WindowBackdropType="Mica"
+    WindowCornerPreference="Round"
+    WindowStartupLocation="CenterScreen"
+    mc:Ignorable="d">
+    <Grid>
+        <Grid.ColumnDefinitions>
+            <ColumnDefinition Width="1*" />
+            <ColumnDefinition Width="4*" />
+        </Grid.ColumnDefinitions>
+        <Grid.RowDefinitions>
+            <RowDefinition Height="Auto" />
+            <RowDefinition Height="10*" />
+        </Grid.RowDefinitions>
+
+        <ui:TitleBar
+            Title="Ruleset Manager - Hurl"
+            Grid.ColumnSpan="2"
+            Icon="pack://application:,,,/Assets/applicationIcon-256.png" />
+
+        <StackPanel Grid.Row="1" Grid.Column="0">
+            <TextBlock FontSize="18" Text="Rulesets:" />
+            <ui:Button HorizontalAlignment="Stretch">Create</ui:Button>
+            <ui:Card VerticalAlignment="Stretch">
+                <StackPanel />
+            </ui:Card>
+        </StackPanel>
+
+        <StackPanel Grid.Row="1" Grid.Column="1">
+            <TextBlock
+                x:Name="WarnText"
+                Height="0"
+                HorizontalAlignment="Center"
+                Foreground="Red"
+                Text="Atleast input Target Browser and Rules fields!"
+                Visibility="Hidden" />
+
+            <ui:Card Padding="10">
+                <Grid>
+                    <Grid.RowDefinitions>
+                        <RowDefinition Height="1*" />
+                        <RowDefinition Height="1*" />
+                        <RowDefinition Height="1*" />
+                    </Grid.RowDefinitions>
+                    <Grid.ColumnDefinitions>
+                        <ColumnDefinition Width="1*" />
+                        <ColumnDefinition Width="150" />
+                    </Grid.ColumnDefinitions>
+
+                    <!--<TextBlock VerticalAlignment="Center" Text="Rule Mode:" /> -->
+
+                    <TextBlock VerticalAlignment="Center" Text="Target Browser:" />
+                    <ComboBox
+                        x:Name="TargetBrowser"
+                        Grid.Column="1"
+                        Margin="0,0,0,10"
+                        ItemsSource="{Binding}" />
+                    <TextBlock
+                        Grid.Row="1"
+                        VerticalAlignment="Center"
+                        Text="Alternate Launch:" />
+                    <ComboBox
+                        x:Name="TargetBrowserx"
+                        Grid.Row="1"
+                        Grid.Column="1"
+                        ItemsSource="{Binding}" />
+                </Grid>
+            </ui:Card>
+
+            <ui:Card>
+                <StackPanel Orientation="Horizontal">
+                    <ComboBox
+                        x:Name="RuleModeInput"
+                        Margin="10"
```

Comment:
Personally, I would move any margin needed to the StackPanel and use `Spacing` for between the elements. Also, it is recommended to always use multiples of four. This is to be safe with display scaling factors (think of 125%)

Reply:
>  always use multiples of four,

Sure....